WinRmClient cleanup

neykov opened the pull request:
A general refactoring, no new features added. Breaks up the huge `WinRmClient` file into several smaller pieces. Easier reviewed by looking at the individual commits.

This has been sitting around in a giant stash since before the 0.5.0 release, but decided it's too much to include in the release. I just split it into smaller, logical commits now. Some more to follow, mostly focusing on streamlining the tests.

bostko commented:
@neykov you deprecated methods and left `io.cloudsoft.winrm4j.winrm.WinRmTool` deprecated code being used.
Could you update WinRmTool or provide example PR for https://github.com/apache/brooklyn-server

aledsage commented:
@neykov have only had a chance for a very quick look, but it looks good - nice to split it up into more manageable classes!

neykov commented:
@bostko addressed comments - fixed failing test case (good catch!) and removed calls to deprecated methods.

As for updating Brooklyn's wrapper - it's perfectly fine to use the existing deprecated methods, but I'll try to update it after we cut a new winrm4j release.

bostko commented on client/src/main/java/io/cloudsoft/winrm4j/client/ShellCommand.java:
```diff
@@ -0,0 +1,212 @@
+    }
+
+    public int getNumberOfReceiveCalls() {
+        return numberOfReceiveCalls;
```
`getNumberOfReceiveCalls` test is failing with different error now.
```
java.lang.NullPointerException
	at io.cloudsoft.winrm4j.client.WinRmClient.getNumberOfReceiveCalls(WinRmClient.java:203)
	at io.cloudsoft.winrm4j.winrm.WinRmTool.executeCommand(WinRmTool.java:248)
```
`shellCommand` is null.
I agree there is no point in a user using the method.
However I do not agree putting it as deprecated.
Polling mechanism for long running commands is relying on proper `OPERATION_TIMEOUT` handling. So we need to test that and not deprecate it.

bostko replied:
I suggest putting getNumberOfReceiveCalls as package private method.
I may do contribution to the PR in that regard but it won't be immediate.

neykov replied:
@bostko fixed this and verified all tests are successful :) (this time after doing all the changes!)
re deprecating `getNumberOfReceiveCalls` - this was my thinking as well, it's an implementation detail, exposed because the tests reference it (and they use it in a heuristic check). Let's keep it deprecated this release so potential users will be notified it's going away and make it accessible only to tests in future. Or figure out an alternative way to verify the behaviour without needing to expose it.

bostko commented:
Tests pass. Good to merge I think.

neykov commented:
Thanks @bostko, merged.
